### PR TITLE
Add tests and bug reports for failing`get-model` with `chop`

### PR DESCRIPTION
**Pull request by palinatolmach**
Blocked on: https://github.com/runtimeverification/haskell-backend/issues/3616

This PR introduces two tests for the `get-model` endpoint that involve `chop` function defined as:

```
  syntax Int ::= chop ( Int )           [function, total, klabel(chop), smtlib(chop)]
  syntax Int ::= "pow256"               [alias] /* 2 ^Int 256 */

  rule pow256 => 115792089237316195423570985008687907853269984665640564039457584007913129639936
  rule chop ( I:Int ) => I modInt pow256 [concrete, smt-lemma]
```

The first test `sat-chop-single-int` shows that the variable substitution returned for `chop(x) == x && x == 1` is correct: `x = 1`. The second, failing, test `sat-chop-sub-ints` shows that the reported substitution (`x = 0`) is incorrect for a slightly more complex expression involving subtraction: `chop(1 - x) == 0`.

The changes in [pyk/testing/_kompiler.py](https://github.com/runtimeverification/pyk/compare/smt-lemma-translation?expand=1#diff-f7eb769a6f82214d26847367701f3229182d9ae9890962becdd1b5af638b3750) add bug-report generation functionality for tests defined in `KoreClientTest`.

### CODE REVIEW

File: src/pyk/testing/_kompiler.py [hunk 1]
```diff
@@ -53,11 +53,11 @@ class KompiledTest:
...
 
-    @pytest.fixture(scope='class')
-    def bug_report(self) -> BugReport | None:
-        return None
+    @pytest.fixture
+    def bug_report(self, tmp_path: Path) -> BugReport:
+        # return None
         # Use the following line instead to generate bug reports for tests
-        # return BugReport(Path('bug_report'))
+        return BugReport(tmp_path / 'bug-report')
```
Comment (ehildenb) [inline src/pyk/testing/_kompiler.py:60]: Is it intended that this in uncommented now? We are going to generate bug reports for every test run, which could be a lot of extra files.

Comment (palinatolmach) [reply src/pyk/testing/_kompiler.py]: It wasn't, sorry, thanks for noticing; didn't realise we're going to merge it. Done in 
https://github.com/runtimeverification/pyk/commit/313ec9512bb7dc280d765247d99cfeef88054eef.

**Comment by palinatolmach** [comment]
As noticed by @jberthold, Z3 returns `x = <value of pow256 + 1>` (instead of a more obvious `x = 1`) as a substitution for `chop(1 - x) == 0`. https://github.com/runtimeverification/pyk/pull/520/commits/2a90d1385dff3e76e7c563b4cdf3b8c53f7904b0 updates the test to match the generated model.

**Review by ehildenb — APPROVED** [review]

File: src/pyk/testing/_kompiler.py [hunk 2]
```diff
@@ -141,6 +141,21 @@ class KoreClientTest(KompiledTest):
...
+        client.close()
+        server.close()
+
+
+class TestKoreClientWithSMTLemmas(KompiledTest):
+    KOMPILE_BACKEND = 'haskell'
+
+    KORE_MODULE_NAME: ClassVar[str]
+    KORE_CLIENT_TIMEOUT: ClassVar = 1000
+
```
Comment (tothtamas28) [inline src/pyk/testing/_kompiler.py:158]: This is probably leftover from local testing.

Comment (palinatolmach) [reply src/pyk/testing/_kompiler.py:158]: To be fair, I copied it from a similar `class KoreClientTest(KompiledTest)`: https://github.com/runtimeverification/pyk/blob/c448f25e78d8bdb21ef4db7d052712edbb51d385/src/pyk/testing/_kompiler.py#L138. Should I have removed it?

**Pull request merged**